serviceLibrary/build.gradle

```diff
@@ -59,6 +59,8 @@ dependencies {
     implementation 'androidx.legacy:legacy-support-v4:1.0.0'
     implementation "androidx.core:core-ktx:1.13.1"
     implementation "org.jetbrains.kotlinx:kotlinx-coroutines-core:1.9.0"
+    implementation "org.jetbrains.kotlinx:kotlinx-coroutines-android:1.9.0"
```

Comment:
Hmmm, other guys don't need it. Why do you need it ?
Do you have an example ?

Reply:
The reason I have added this dependency kotlinx-coroutines-android:1.9.0 is to support the Main Dispatcher, which is required to launch coroutines on the main thread. Specifically, the main thread is crucial when you're updating UI elements or performing any task that interacts with the Android UI.

Without this ,

In your case, in MqttConnection.kt, we have:


CoroutineScope(Dispatchers.Main).launch {
    Timber.e(ex, "Exception occurred attempting to reconnect: ${ex.message}")
    setConnectingState(false)
    handleException(resultBundle, ex)
}

Here, Dispatchers.Main refers to the Main Dispatcher, which is responsible for executing the coroutine on the Android main (UI) thread. The kotlinx-coroutines-android library provides this dispatcher, allowing you to launch coroutines that interact with the UI, like logging, setting connection state, or handling exceptions in this case.

Without kotlinx-coroutines-android, the Main dispatcher would not be available, and you would get an error because Android needs this dispatcher to work with UI elements safely.